### ISSUE TITLE
Lesson Contribution - Additional Commands

### ISSUE THREAD

**Issue by Talishask** [swcarpentry/git-novice#765]
I'm a member of The Carpentries staff and I'm submitting this issue on behalf of another member of the community. In most cases, I won't be able to follow up or provide more details other than what I'm providing below.

---
Useful git option when fiddling inside local repository doing experiments but in the end we do not want to push these changes. Instead we want to pull again the code from the repository as it was before we started experimenting. This could be included as an exercise or a useful tip (I have this situation quite often since I do lots of experimenting when coding and some of the changes turn out to be dead end so I just want to get git repo without the need to download everything from scratch)
The command is 
**git fetch –all**        (this will remove all local commits that have not been pushed)
and then 
**git reset –hard origin/master**
OR if some other branch is wanted
**git reset –hard origin/<branch_name>**
**_Explanation:_**
git fetch downloads the latest from remote without trying to merge or rebase anything.
Then the git reset resets the master branch to what you just fetched. The --hard option changes all the files in your working tree to match the files in origin/master.
Moreover if we would like to maintain the local commits we can do that by creating a new branch before resetting:
**git checkout master
git branch new-branch-for-preserving-current-commits
git fetch –all
git reset –hard origin/master**
( uncommitted changes will be lost – so make sure to stach and commit anything that is needed)

**Comment by kekoziar** [comment]
Hi, @Talishask.  I absolutely agree that these are useful commands for advanced users of git.  I actually use the quite a lot as a maintainer of this lesson because I work with multiple PRs and repositories.  However, they are advanced git, and out of scope of this lesson.  I encourage anyone who wants to tackle an advanced git lesson to refer to this.
